minimal-shadow-cljs-nodejs/src/server/main.cljs

```diff
@@ -132,78 +133,98 @@
                                           :submit {:type "plain_text",
                                                    :text "Submit"}))
-      
+
       ;; TODO FIXME
       #_"broadcast2:channel-select"
-      #_(slack/views-update! (j/get-in payload ["container" "view_id"])
-                             (assoc (modal-view-payload "Compose Broadcast" blocks-broadcast-2)
-                                    :submit {:type "plain_text", :text "Submit"})))
+      #_(slack/views-push! (j/get-in payload ["container" "view_id"])
+                           (assoc (modal-view-payload "Compose Broadcast" blocks-broadcast-2)
+                                  :submit {:type "plain_text", :text "Submit"})))
 
-    "view_submission" ; "Submit" button pressed
+    "view_submission"                                       ; "Submit" button pressed
     ;; In the future we will need to branch on other data
-    (do (request-updates! (decode-text-input (j/get-in payload ["view"
-                                                                "state"
-                                                                "values"
-                                                                "sb-input1"
-                                                                "broadcast2:text-input"
-                                                                "value"]))
-                          (map :id (:slack/channels-raw @db)))
-        (callback nil #js {:statusCode 200}))))
+    (p/let [channel-ids (p/-> (slack/get+ "channels.list")
+                              (j/get :channels)
+                              (->> (keep (j/fn [^:js {:keys [is_member id]}]
+                                           ;; TODO
+                                           ;; ensure bot joins team-channels when they are created
+                                           (when is_member id)))))
+            message-text (decode-text-input (get-in payload [:view
+                                                             :state
+                                                             :values
+                                                             :sb-input1
+                                                             :broadcast2:text-input
+                                                             :value]))]
+      (request-updates! message-text channel-ids))))
 
 
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 ;;;; SparkBoard SlackBot server
 
-(defn response [cb body]
-  (cb nil #js {:statusCode 200 :body body}))
+(defn response [body]
+  (p/let [body-resolved body]
+    (println :responding-with body-resolved)
+    (p/resolve #js {:statusCode 200 :body (clj->js body-resolved)})))
+
+(j/defn parse-body [^:js {:as event
+                          :keys [body]
+                          {:keys [Content-Type]} :headers}]
+  (case Content-Type
+    "application/x-www-form-urlencoded" (uri/query-string->map (do #_decode-base64 body))
```

Comment:
I'm quite surprised that the global shortcut works without base64 decoding. I'm not going to test right now because I don't want to upload tests to AWS before our demo and I still need to catch up with the local testing config. But were you not receiving payloads like "cGF5bG9hZD0lN0IlMjJ0eXBlJTIyJTNBJTIyc2hvcnRjdXQlMjIlMkMlMjJ0b2tlbiUyMiUzQSUyMjJHSTVkSHNOYWJ4ZmZLc2N2eEszbkJ3aCUyMiUyQyUyMmFjdGlvbl90cyUyMiUzQSUyMjE1ODk1NTI5NDEuMjQ0ODQ4JTIyJTJDJTIydGVhbSUyMiUzQSU3QiUyMmlkJTIyJTNBJTIyVDAxME1HVlQ0VFYlMjIlMkMlMjJkb21haW4lMjIlM0ElMjJzcGFya2JvYXJkLWFwcCUyMiU3RCUyQyUyMnVzZXIlMjIlM0ElN0IlMjJpZCUyMiUzQSUyMlUwMTJFNDgwTlRCJTIyJTJDJTIydXNlcm5hbWUlMjIlM0ElMjJkYXZlLmxpZXBtYW5uJTIyJTJDJTIydGVhbV9pZCUyMiUzQSUyMlQwMTBNR1ZUNFRWJTIyJTdEJTJDJTIyY2FsbGJhY2tfaWQlMjIlM0ElMjJzcGFya2JvYXJkJTIyJTJDJTIydHJpZ2dlcl9pZCUyMiUzQSUyMjExMzk0Mzg2ODYzMDUuMTAyMTU3MzkyMjk0Ny5kNDhhNDExNDBmMjJhMjc4YTlmNGUxZTVkNDliMDBjYSUyMiU3RA=="? I definitely was, and it definitely didn't work without base64 decoding.

Reply:
yeah I have no idea where those came from. but there is enough weirdness in their API that I am not surprised. maybe we leave it like this until we find something that breaks, and then we know?